autogpt_platform/frontend/src/components/integrations/credentials-input.tsx

```diff
@@ -246,6 +256,31 @@ export const CredentialsInput: FC<{
         )}
       </>
     );
+    // Only one saved credential
+  } else if (
+    savedApiKeys.length === 0 &&
+    savedOAuthCredentials.length === 1 &&
+    selectedCredentials === undefined
+  ) {
+    if (selectedCredentials === undefined) {
+      onSelectCredentials({
+        id: savedOAuthCredentials[0].id,
+        type: "oauth2",
+        provider,
+        title: savedOAuthCredentials[0].title,
+      });
+    }
+    return null;
+  } else if (savedApiKeys.length === 1 && savedOAuthCredentials.length === 0) {
+    if (selectedCredentials === undefined) {
+      onSelectCredentials({
+        id: savedApiKeys[0].id,
+        type: "api_key",
+        provider,
+        title: savedApiKeys[0].title,
+      });
+    }
+    return null;
```

Comment:
What's the purpose of this? This whole chain of logic seems questionably necessary.

Reply:
Reading through this PR, I don't really see why this shortcut is being taken. What's the big hurdle preventing the regular credentials input from working on a multi-provider block?

Reply:
I'm not sure I understand @Pwuts. The purpose of this code is to choose the only one existing credentials and then not show dropdown at all. I simplified this code.

Reply:
My question is: would it break anything to allow users the choice of adding their own API key? I understand this will be resolved in a follow-up PR, but I don't immediately see what would break if you still showed the credentials picker in all cases.